gymnasium/envs/registration.py

```diff
@@ -521,7 +715,7 @@ def register(
 
 
 def make(
```

Comment:
Personally and I think Ariel agrees, `make` is already too complex and this adds more weird complex code. 
Could we split this into other functions to simplify the overall function.

Reply:
PR is incomplete but I gave this a go (check recent changes)